src/snowflake/snowpark/dataframe.py

```diff
@@ -1089,6 +1090,40 @@ def col(self, col_name: str) -> Column:
         else:
             return Column(self._resolve(col_name))
 
+    def _internal_col_regex(
+        self,
+        regex: str,
+        translate_columns: Optional[Dict[str, str]] = None,
+    ) -> Column:
+        if not isinstance(regex, str):
+            raise ValueError(
+                f"regex provided to col_regex() must be string, got {type(regex)} with value of {regex} instead."
+            )
+        expressions = []
+        if translate_columns:
+            for key, value in translate_columns.items():
+                if re.match(regex, unquote_if_quoted(key)):
+                    expressions.append(self._resolve(value))
+        else:
+            for column in self._output:
+                if re.match(regex, unquote_if_quoted(column.name), flags=re.IGNORECASE):
+                    expressions.append(column)
+        if not expressions:
+            raise ValueError(f"No column match provided regex:{regex}")
+        return Column(Star(expressions))
+
+    def col_regex(self, regex: str) -> Column:
+        """Selects column based on the column name specified as a regex and returns a reference to  it
+        Args:
```

Comment:
I think we need to consider case sensitivity. I know flake unquoted column names are case insensitive but quoted names are sensitive. Do we have test coverage for these scenarios.

Reply:
will add thoses tests